xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-api/src/main/java/org/xwiki/ratings/internal/SolrRatingsManager.java

```diff
@@ -563,4 +567,30 @@ public AverageRating recomputeAverageRating(EntityReference entityReference) thr
             throw new RatingsException(AVERAGE_RATING_NOT_ENABLED_ERROR_MESSAGE);
         }
     }
+
+    private void checkIfVoteValid(int vote) throws RatingsException
+    {
+        // If the vote is outside the scope of the scale, we throw an exception immediately.
+        if (vote < 0 || vote > this.getScale()) {
+            throw new RatingsException(String.format("The vote [%s] is out of scale [%s] for [%s] rating manager.",
+                vote, this.getScale(), this.getIdentifier()));
+        }
+    }
+
+    private void checkIfDocumentExists(EntityReference reference) throws RatingsException
+    {
+        try {
+            EntityReference entityReference = reference.extractReference(DOCUMENT);
+            if (entityReference == null || !this.documentAccessBridge.exists((DocumentReference) entityReference)) {
```

Comment:
Assuming `extractDocument` always return `null` or a `DocumentReference`

Reply:
An EntityReference of type DOCUMENT is not necessarily a DocumentReference so you should never assume this IMO.

Reply:
Thanks for confirming. I've seen `this.documentAccessBridge.exists(new DocumentReference(entityReference))` used in our codebase.
Unless we have a more generic API to check if an entity reference exists.

Reply:
> Unless we have a more generic API to check if an entity reference exists.

You could add one.

Reply:
I added a new API `DocumentReference.valueOf(EntityReference)` a more generic API for `exists(EntityReference)` is a bit more complex as it'd need to handle the various cases of checking if a wiki exists, if a property exists etc. And we'd need to decide where exactly to put it.